dasharo-security/cbnt.robot: add test cases

filipleple opened the pull request:
Add cases for more requirements that can be verified based on cbmem output:

- TPM Startup from locality 3
- Fused platform EoM set and FPFs Committed

*ref: ncm-1813*

filipleple committed:
dasharo-security/cbnt.robot: add test cases

Add cases for more requirements that can be verified based on cbmem
output:

- TPM Startup from locality 3
- Fused platform EoM set and FPFs Committed

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

SergiiDmytruk commented on dasharo-security/cbnt.robot:
```diff
@@ -91,8 +91,47 @@ CBNT003.201 Converged Boot Guard and TXT - PCR-0 is reconstructed correctly (Ubu
 *** Keywords ***
+Check TPM Startup from locality 3
+    [Documentation]    Check if F, V and M components of the boot policy match
+    ...    profile 5
```
Wrong documentation in both new keywords.

filipleple replied:
[fixed](https://github.com/Dasharo/open-source-firmware-validation/pull/984/files#diff-d8efc5815651fd1ef1282cb97483ee318544e0b21b12a4f1191c13200652220aR114)

SergiiDmytruk commented on dasharo-security/cbnt.robot:
```diff
@@ -91,8 +91,47 @@ CBNT003.201 Converged Boot Guard and TXT - PCR-0 is reconstructed correctly (Ubu
+    [Arguments]    ${os_id}
+    Boot OS And Enter Root Shell    ${os_id}
+    ${out_cbmem}=    Execute Command In Terminal    cbmem -l
+    Should Contain    ${out_cbmem}    TPM Startup locality:    3
```
Like in `Check CBnT Profile 5`:
```suggestion
    Should Match Regexp    ${out_cbmem}    TPM Startup locality:\\S+3\\n
```

filipleple replied:
[fixed](https://github.com/Dasharo/open-source-firmware-validation/pull/984/files#diff-d8efc5815651fd1ef1282cb97483ee318544e0b21b12a4f1191c13200652220aR119)

SergiiDmytruk commented on dasharo-security/cbnt.robot:
```diff
@@ -91,8 +91,47 @@ CBNT003.201 Converged Boot Guard and TXT - PCR-0 is reconstructed correctly (Ubu
+    [Arguments]    ${os_id}
+    Boot OS And Enter Root Shell    ${os_id}
+    ${out_cbmem}=    Execute Command In Terminal    cbmem -l
+    Should Contain    ${out_cbmem}    Manufacturing Mode          : NO
+    Should Contain    ${out_cbmem}    FPFs Committed              : YES
```
The checks could similarly use `Should Match Regexp` to not be too fragile.

filipleple replied:
[fixed](https://github.com/Dasharo/open-source-firmware-validation/pull/984/files#diff-d8efc5815651fd1ef1282cb97483ee318544e0b21b12a4f1191c13200652220aR129)

SergiiDmytruk commented on dasharo-security/cbnt.robot:
```diff
@@ -91,8 +91,47 @@ CBNT003.201 Converged Boot Guard and TXT - PCR-0 is reconstructed correctly (Ubu
+    [Tags]    robot:private
+    [Arguments]    ${os_id}
+    Boot OS And Enter Root Shell    ${os_id}
+    ${out_cbmem}=    Execute Command In Terminal    cbmem -l
```
I think you meant this (and in the other keyword as well):
```suggestion
    ${out_cbmem}=    Execute Command In Terminal    cbmem -1
```

filipleple replied:
:man_facepalming: [fixed](https://github.com/Dasharo/open-source-firmware-validation/pull/984/files#diff-d8efc5815651fd1ef1282cb97483ee318544e0b21b12a4f1191c13200652220aR128)

SergiiDmytruk reviewed (changes requested):
Looks like you haven't run these tests yet :)

filipleple committed:
dasharo-security/cbnt.robot: fix regexp to work with blank spaces

Using S+1 means "any number of NON-whitespace characters before 1".
Changed to use lowercase s, which allows whitespace characters as well,
appropriately for the case we're parsing.

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple commented:
Alright now it's properly fixed and tested. I've used this diff to test it on an unfused&unprovisioned platform:

```
commit 3858eecc3c8de8dc6c62adf3fb89b247e41e6e66 (HEAD -> ibg-more-tests)
Author: Filip Lewiński <filip.lewinski@3mdeb.com>
Date:   Mon Aug 11 11:47:02 2025 +0200

    TEMP: add flags and flip conditions to make it run and pass

    Diff for running and testing on unprovisioned VP2440. The point is to
    check whether the option values are extracted and compared correctly in
    the new test cases, hence the flipped conditions.

    Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

diff --git a/dasharo-security/cbnt.robot b/dasharo-security/cbnt.robot
index 9a78b1be1c92..91c85da8f06b 100644
--- a/dasharo-security/cbnt.robot
+++ b/dasharo-security/cbnt.robot
@@ -32,6 +32,7 @@ CBNT001.201 Converged Boot Guard and TXT - CBnT profile is 5 / FVME (Ubuntu)
     [Documentation]    CBnT profile MUST be 5 - FVME
     Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    CBNT001.201 not supported
     Skip If    '${ENV_ID_UBUNTU}' not in ${TESTED_LINUX_DISTROS}    CBNT001.201 not supported
+    Skip
     Check CBnT Profile 5    ${ENV_ID_UBUNTU}

 CBNT002.101 Converged Boot Guard and TXT Status Menu is visible
@@ -59,6 +60,7 @@ CBNT003.201 Converged Boot Guard and TXT - PCR-0 is reconstructed correctly (Ubu
     ...    log the data CBnT used to extend PCR-0
     Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    CBNT003.201 not supported
     Skip If    '${ENV_ID_UBUNTU}' not in ${TESTED_LINUX_DISTROS}    CBNT003.201 not supported
+    Skip

     Boot OS And Enter Root Shell    ${ENV_ID_UBUNTU}

@@ -127,8 +129,8 @@ Check EoM And FPFs Committed
     [Arguments]    ${os_id}
     Boot OS And Enter Root Shell    ${os_id}
     ${out_cbmem}=    Execute Command In Terminal    cbmem -1 | grep ME
-    Should Match Regexp    ${out_cbmem}    Manufacturing Mode\\s+:\\s+NO\n
-    Should Match Regexp    ${out_cbmem}    FPFs Committed\\s+:\\s+YES\n
+    Should Match Regexp    ${out_cbmem}    Manufacturing Mode\\s+:\\s+YES\n
+    Should Match Regexp    ${out_cbmem}    FPFs Committed\\s+:\\s+NO\n
     Exit From Root User

 Check CBnT Profile 5
diff --git a/platform-configs/protectli-vp2440.robot b/platform-configs/protectli-vp2440.robot
index 3e7bf02fe61d..afce213daacc 100644
--- a/platform-configs/protectli-vp2440.robot
+++ b/platform-configs/protectli-vp2440.robot
@@ -54,3 +54,5 @@ ${SATA_SUPPORT}=                    ${TRUE}
 ${CLEVO_DISK}=                      Protectli 256GB M.2
 ${DEVICE_NVME_DISK}=
 ...                                 Non-Volatile memory controller: Phison Electronics Corporation PS5019-E19 PCIe4 NVMe
 Controller
+${INTEL_CBNT_SUPPORT}=                  ${TRUE}
+${INTEL_CBNT_BOOTGUARD_FUSED}=                      ${TRUE}
```

Log attached:
[cbnt_log.tar.gz](https://github.com/user-attachments/files/21713016/cbnt_log.tar.gz)

filipleple committed:
dasharo-security/cbnt.robot: fix test case documentation

Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>
